main.js

```diff
@@ -0,0 +1,159 @@
+const VALIDATE = (prop) => {
+    return prop === undefined || prop === null || prop === "";
+}
+
+const validateNumbers = (prop) => {
+    prop === NaN;
+}
+
+// 1. Створити масив, довжину та елементи якого задає користувач (через prompt). Елементами масиву повинні бути числа.
+const arraySize = prompt("Please, enter an array length:");
+let newArray = [];
+
+const createArray = (size, arr) => {
+    if(VALIDATE(size) || size < 1 || validateNumbers(size)) {
+        prompt("Please try one more time and enter the length of your future array!");
+    } else {
+        for (i = 0; i < size; i++) {
+            arr[i] = prompt("Please enter a number.");
+                if(VALIDATE(arr[i]) || validateNumbers(arr[i])) {
+                    return;
+                }
+            }
+        } console.log(arr);
+}
+createArray(arraySize, newArray);
+
+// 2. Відсортувати масив за зростанням.
+newArray.sort((num_1, num_2) => {
+    return num_1 - num_2;
+});
+console.log(newArray);
+
+
+// 3. Дано масив - список покупок. Кожен елемент масиву - це обʼєкт вигляду: {productName: 'bread', productPrice: 14.5, productQuantity: 2}. Мінімальний довжина такого масиву - 6
+
+const productArray = [
+    {productName: "молоко", productPrice: 35, productQuantity: 2},
+    {productName: "сир", productPrice: 100, productQuantity: 5},
+    {productName: "хліб", productPrice: 35, productQuantity: 4},
+    {productName: "м'ясо", productPrice: 150, productQuantity: 3},
+    {productName: "виноград", productPrice: 85, productQuantity: 1},
+    {productName: "авокадо", productPrice: 60, productQuantity: 4},
+    {productName: "рисова локшина", productPrice: 200, productQuantity: 2},
+];
+// Created a function to sort the array by field.
+const sortedByField = (field) => {
+    return (a, b) => a[field] > b[field] ? 1 : -1;
+};
+
+// Порахувати загальну вартість корзини та вивести суму у зрозумілому форматі
+const totalPrice = productArray.reduce((totalPrice, {productPrice, productQuantity}) => totalPrice + productPrice * productQuantity, 0);
+console.log(`Загальна вартість продуктової корзини складає ${totalPrice} грн.`);
+
+// Порахувати найменшу кількість продукту, який потрібно купити
+const sortedByQuantity = productArray.sort(sortedByField("productQuantity"));
+console.log(sortedByQuantity)
+// console.log("Найменша кількість продукту, який ми маємо придбати - " + sortedByQuantity[0].productName + " - " + sortedByQuantity[0].productQuantity + ".")
+
+// Порахувати загальну кількість продуктів
+const totalQantity = productArray.reduce((totalQantity, {productQuantity}) => totalQantity + productQuantity, 0);
```

Comment:
```suggestion
const totalQuantity = productArray.reduce((totalQuantity, {productQuantity}) => totalQuantity + productQuantity, 0);
```

Reply:
Can you please explain what I have to change in my code? In "suggested changes" I see two similar strings.

Reply:
Its typo there `totalQantity` => `totalQuantity`